Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,560 @@
+        thetax_std = std::sqrt(thetax2_ave - thetax_ave*thetax_ave);
+
+        m_data[get_idx("x_ave_"+species_names[i_s])] = x_ave;
+        m_data[get_idx("x_std_"+species_names[i_s])] = x_std;
+        m_data[get_idx("thetax_min_"+species_names[i_s])] = thetax_min;
+        m_data[get_idx("thetax_ave_"+species_names[i_s])] = thetax_ave;
+        m_data[get_idx("thetax_max_"+species_names[i_s])] = thetax_max;
+        m_data[get_idx("thetax_std_"+species_names[i_s])] = thetax_std;
+
+#elif defined(WARPX_DIM_3D)
+        // w_tot
+        // x_ave, x^2_ave, y_ave, y^2_ave,
+        // thetax_min, thetax_ave, thetax^2_ave, thetax_max
+        // thetay_min, thetay_ave, thetay^2_ave, thetay_max
+        amrex::Real w_tot = 0.0_rt;
+        amrex::Real x_ave = 0.0_rt;
+        amrex::Real x2_ave = 0.0_rt;
+        amrex::Real y_ave = 0.0_rt;
+        amrex::Real y2_ave = 0.0_rt;
+        amrex::Real thetax_min = 0.0_rt;
+        amrex::Real thetax_ave = 0.0_rt;
+        amrex::Real thetax2_ave = 0.0_rt;
+        amrex::Real thetax_max = 0.0_rt;
+        amrex::Real thetay_min = 0.0_rt;
+        amrex::Real thetay_ave = 0.0_rt;
+        amrex::Real thetay2_ave = 0.0_rt;
+        amrex::Real thetay_max = 0.0_rt;
+
+        amrex::ReduceOps<ReduceOpSum,
+                         ReduceOpSum, ReduceOpSum, ReduceOpSum, ReduceOpSum,
+                         ReduceOpMin, ReduceOpSum, ReduceOpSum, ReduceOpMax,
+                         ReduceOpMin, ReduceOpSum, ReduceOpSum, ReduceOpMax> reduce_ops;
+        auto r = amrex::ParticleReduce<amrex::ReduceData<Real,
+                                                         Real, Real, Real, Real,
+                                                         Real, Real, Real, Real,
+                                                         Real, Real, Real, Real>>(
+            myspc,
+            [=] AMREX_GPU_DEVICE(const PType& p) noexcept -> amrex::GpuTuple<Real,
+                                                                             Real, Real, Real, Real,
+                                                                             Real, Real, Real, Real,
+                                                                             Real, Real, Real, Real>
+            {
+                const amrex::Real w  = p.rdata(PIdx::w);
+                const amrex::Real x = p.pos(0);
+                const amrex::Real y = p.pos(1);
+                const amrex::Real ux = p.rdata(PIdx::ux);
+                const amrex::Real uy = p.rdata(PIdx::uy);
+                const amrex::Real uz = p.rdata(PIdx::uz);
+                const amrex::Real thetax = std::atan2(ux, uz);
+                const amrex::Real thetay = std::atan2(uy, uz);
+                return {w, w*x, w*x*x, w*y, w*y*y,
+                        thetax, w*thetax, w*thetax*thetax, thetax,
+                        thetay, w*thetay, w*thetay*thetay, thetay};
+            },
+            reduce_ops);
+
+        w_tot  = amrex::get<0>(r);
```

Comment:
The code could be simplified by doing the declaration here and removing the ones above
```amrex::Real w_tot = amrex::get<0>(r);```